code/controllers/ErrorPageControllerExtension.php

```diff
@@ -9,7 +9,8 @@
 class ErrorPageControllerExtension extends Extension {
 
 	public function onBeforeHTTPError($statusCode, $request) {
-		$response = ErrorPage::response_for($statusCode);
+		// Don't use the HTML response for media requests
+		$response = $request->isMedia() ? null : ErrorPage::response_for($statusCode);
```

Comment:
Yeah I agree full 404 response is better and consistent with current behaviour so I'll remove that check.